fontir/src/stateset.rs

```diff
@@ -79,23 +103,31 @@ impl FileStateSet {
     }
 
     /// Pay attention to path, we'd like to know if it changes.
-    pub fn insert(&mut self, path: &Path) -> Result<(), io::Error> {
-        if !self.entries.contains_key(path) {
-            self.entries
-                .insert(path.to_path_buf(), FileState::of(path)?);
-
-            // For a dir to register unchanged we need to add it's current contents
-            if path.is_dir() {
-                let mut dirs_visited = HashSet::new();
-                for new_path in self.new_files(&mut dirs_visited, path) {
-                    self.entries
-                        .insert(new_path.clone(), FileState::of(&new_path)?);
-                }
+    pub fn track_file(&mut self, path: &Path) -> Result<(), io::Error> {
+        self.entries
+            .insert(path.to_path_buf(), State::File(FileState::of(path)?));
+
+        // For a dir to register unchanged we need to add it's current contents
+        if path.is_dir() {
+            let mut dirs_visited = HashSet::new();
+            for new_path in self.new_files(&mut dirs_visited, path) {
+                self.entries
+                    .insert(new_path.clone(), State::File(FileState::of(&new_path)?));
             }
         }
         Ok(())
     }
 
+    /// Pay attention to slice, we'd quite like to know if it changes
+    ///
+    /// Path is used only as an identifier, invent any scheme you like.
+    /// my_file/my_subsection say.
```

Comment:
> Path is used only as an identifier

then why do you use `Path` and not simply a string?

Reply:
it doesn't seem like you're using any Path-specific methods for the SliceSet. 
Path/PathBuf are wrappers around OsString/OsStr so on windows they will use backslashes and do other win-specific things. Any methods that attempts to access the underlying filesystem is bound to fai..
It feels weird to me. If there was something like python's pathllib.PurePath maybe.. but it doesn't seem to be the case in rust.
I think it'd be clearer if SliceSet used a String for the entries key.

Reply:
I view Path/PathBuf as equivalent to Python's Path, is that wrong?

Reply:
yes. pathlib.Path is same as rust's Path/PathBuf, but it seems like there's no equivalent in Rust for a [PurePath](https://docs.python.org/3/library/pathlib.html#pure-paths), ie. "objects provide path-handling operations which don’t actually access a filesystem."

Reply:
A Path you never use for file IO is?

But I think you are right, this doesn't need to be a Path. I made it a String.

Reply:
> A Path you never use for file IO is?

it's a string that has additional path-like methods to compose, split, etc. like a path, very convenient at times.